Adding tests for console, ip, setup-cdk commands, dns, better ssh solution

agajdosi opened the pull request:
Minishift var changed to be global.
In php and nodejs changed tempalte parameter to template.
Upgraded ssh test to test with "whoami", not return message of ssh, which is different on docker, centos-iso and rhel-iso.
Added support for --iso-url flag in minishift start - as we do not know, how exactly will start be implemented in CDK builds.

Commit:
minishift variable needs to be global

Commit:
improvement of ssh test

SSH test now runs "whoami" in box to check if ssh was successful. In
previous solution ssh command was expecting "Last login:" but this
worked only with centos image, not with native boot2docker.

Commit:
test_ms_start support for --iso-url flag

Commit:
disable openshift tests in production branch

Commit:
set ms_start timeout quite loooong

Commit:
minishift start timeout to 5 mins

Commit:
make it more verbose

Commit:
Typos in parameters, changes tempalte to template

Commit:
Adding test case for "minishift console"

This test mimics user who after start of minishift wants to open openshift web console.
First she enters "minishift console --url" to get url of openshift.
Then she access this url. Due to problems with redirection and curl command, this test adds /console/ to the end of returned url.

Commit:
Merge pull request #1 from agajdosi/master

test case for openshift console

Commit:
disabling console test

Commit:
start timeout to 10mins, enabled reuse test

Commit:
Update minishift_sanity.py

Commit:
Updating for CDK3

Commit:
Merge pull request #2 from agajdosi/master

Updating for CDK3

Commit:
Fixing mistake in expected delete response

Commit:
Merge pull request #3 from agajdosi/master

Fixing mistake in expected delete response

Commit:
Update for cdk3 nightly, new tests

Commit:
Merge branch 'production' into master

Commit:
Merge pull request #4 from agajdosi/master

Update for cdk3 nightly, new tests

amitkrout committed:
Merge branch 'master' into master

Commit:
fixing indentation

Commit:
again indentation

(work with whitespaces is really bad on github...)

amitkrout commented on config/config.yaml:
```diff
@@ -1,7 +1,6 @@
-minishift_lib_MODULE : ""
+Hypervisor_Provider : "kvm"
+minishift_PATH : "/home/agajdosi/minishift/minishift"
+minishift_lib_MODULE : "/home/agajdosi/git/cdk-minishift-testsuite/library/minishift_testlib.py"
```
Make these dynamic values generic such as "". These values should be loaded just before the script runs. By using sed command it can be achieved.

agajdosi replied:
Agree, fixed in https://github.com/agajdosi/cdk-minishift-testsuite/commit/a65f445d540ea81d75f1013a14af36db871f6731

amitkrout commented on minishift_sanity.py:
```diff
@@ -92,90 +92,164 @@ def test_ms_start(self):
-        else:
-            cmd = "minishift start --vm-driver " +self.Hypervisor_Provider +" --iso-url " +self.iso_url
-        self.log.info("Starting minishift...")
+	if self.iso_url:
```
How it handles if i am giving the url in the config file. For example if no url is given the cmd will be 
cmd=minishift start --show-libmachine-logs --v=5 --iso-url 
else for any url cmd would be  cmd=minishift start --show-libmachine-logs --v=5 --iso-url xxxxxxxxx. So i dont think it would go to else loop.

agajdosi replied:
Python handles "" in if conditional as false result.
```
if "":
    print "it is true"
else:
    print "not true"
>> not true
```
I am using it like this quite often but 1. I am not sure if it is ok with PEP-8 and 2. not sure if it is widely known, so I can change it to self.iso_url == "" which is maybe more readable, WDYT?

amitkrout replied:
Even i am not sure on PEP-8 and 2. I think this would be more readable/understandable rather self.iso_url=="" and moreover code wise its looks better.

amitkrout commented on minishift_sanity.py:
```diff
@@ -92,90 +92,164 @@ def test_ms_start(self):
+    """
     
-    """        
+    def test_ms_setup_cdk(self):
```
What is the need of this method ?

agajdosi replied:
Tests new downstream command which installs RHEL iso, ocp binary and config minishift to use them natively. Will rewrite this to be able to run also in upstream.

agajdosi replied:
Added Is_Downstream boolean variable into yaml config. If set to true, test for downstream setup-cdk command will run, if not it will be marked passed and rest of test is skipped.

Commit:
deleted pre-filled values in config.yaml

Commit:
deleting prefilled values which I added by mistake

agajdosi commented on minishift_sanity.py:
```diff
@@ -92,90 +92,164 @@ def test_ms_start(self):
-        else:
-            cmd = "minishift start --vm-driver " +self.Hypervisor_Provider +" --iso-url " +self.iso_url
-        self.log.info("Starting minishift...")
+	if self.iso_url:
```
Agree, it was mistake, fixing it in https://github.com/amitkrout/cdk-minishift-testsuite/pull/1/commits/a65f445d540ea81d75f1013a14af36db871f6731.

amitkrout commented on minishift_sanity.py:
```diff
@@ -92,90 +92,164 @@ def test_ms_start(self):
-        if iso_url == "":
-            cmd = "minishift start --vm-driver " +self.Hypervisor_Provider
-        else:
-            cmd = "minishift start --vm-driver " +self.Hypervisor_Provider +" --iso-url " +self.iso_url
```
minishift with rhel iso requires RHN credentials so the command would be look like this  

cmd = "minishift start --vm-driver " +self.Hypervisor_Provider +" --iso-url " +self.iso_url "--username " +self.self.params.get('RHN_USERNAME') +"  --password  " +self.params.get('RHN_PASSWORD')

NOTE :  RHN_USERNAME and RHN_PASSWORD are set in config.yaml

IMO its beter to have one more "else if" condition. Feel free to comment

Commit:
fixing search for oc binary, updating yaml config

Commit:
Merge branch 'production' into master

Commit:
Merge pull request #5 from agajdosi/master

updating setup-cdk test

Commit:
adding sleep to start and stop

Commit:
more delays

Commit:
Better logging, adding delays to start and stop commands

Commit:
Merge branch 'production' into master

Commit:
fixing that child.before() returns type not string if empty

When error occurs child.before() is empty sometimes and returns type not
string which leads to an error when putting message to log. Now
child.before() is in str() to prevent this behaviour.
